pkg/client/cache/listers.go

```diff
@@ -106,6 +107,44 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 	return machines, nil
 }
 
+func (s *StoreToNodeLister) NodeCondition(conditionType api.NodeConditionType, conditionStatus api.ConditionStatus) storeToNodeConditionLister {
```

Comment:
I think my preference would be to do the filtering in the scheduler, but I guess this is ok too. Perhaps the name "FilterConditions"?

Also add a // TODO: make this into a field selector query so the work can be pushed down the stack into apiserver.